code/modules/surgery/bodyparts/_bodyparts.dm

```diff
@@ -1134,3 +1134,27 @@
 		owner.update_body_parts()
 	else
 		update_icon_dropped()
+
+/obj/item/bodypart/emp_act(severity)
+	. = ..()
+	if(. & EMP_PROTECT_WIRES || !(bodytype & BODYTYPE_ROBOTIC))
+		return FALSE
+	owner.visible_message(span_danger("[owner]'s [src.name] seems to malfunction!"))
+
+	var/time_needed = 4 SECONDS
+	var/brute_damage = 2.5
+	var/burn_damage = 7.5
+
+	if(severity == EMP_HEAVY)
+		time_needed *= 2
+		brute_damage *= 2
+		burn_damage *= 2
+
+	receive_damage(brute_damage, burn_damage)
+	do_sparks(number = 1, cardinal_only = FALSE, source = owner)
+	ADD_TRAIT(src, TRAIT_PARALYSIS, "idk")
```

Comment:
> "idk"

change this

Reply:
idk to what trait source

Reply:
EMP_TRAIT? come on man it gotta be something